provider/cloudflare/cloudflare.go

```diff
@@ -359,6 +375,77 @@ func (p *CloudFlareProvider) ApplyChanges(ctx context.Context, changes *plan.Cha
 	return p.submitChanges(ctx, cloudflareChanges)
 }
 
+// submitCustomHostnameChanges implements Custom Hostname functionality for the Change, returns false if it fails
+func (p *CloudFlareProvider) submitCustomHostnameChanges(ctx context.Context, zoneID string, change *cloudFlareChange, chs CustomHostnamesMap, logFields log.Fields) bool {
+	failedChange := false
+	// return early if disabled
+	if !p.CustomHostnamesConfig.Enabled {
+		return !failedChange
+	}
+
+	switch change.Action {
+	case cloudFlareUpdate:
+		if recordTypeCustomHostnameSupported[change.ResourceRecord.Type] {
+			prevChName := change.CustomHostnamePrev
+			newChName := change.CustomHostname.Hostname
+			if prevCh, err := getCustomHostname(chs, prevChName); err == nil {
+				prevChID := prevCh.ID
+				if prevChID != "" && prevChName != newChName {
+					log.WithFields(logFields).Infof("Removing previous custom hostname %q/%q", prevChID, prevChName)
+					chErr := p.Client.DeleteCustomHostname(ctx, zoneID, prevChID)
+					if chErr != nil {
+						failedChange = true
+						log.WithFields(logFields).Errorf("failed to remove previous custom hostname %q/%q: %v", prevChID, prevChName, chErr)
+					}
+				}
+			}
+			if newChName != "" {
+				if prevChName != newChName {
+					log.WithFields(logFields).Infof("Adding custom hostname %q", newChName)
+					_, chErr := p.Client.CreateCustomHostname(ctx, zoneID, change.CustomHostname)
+					if chErr != nil {
+						failedChange = true
+						log.WithFields(logFields).Errorf("failed to add custom hostname %q: %v", newChName, chErr)
+					}
+				}
+			}
```

Comment:
```suggestion
			if newChName != "" && prevChName != newChName {
				log.WithFields(logFields).Infof("Adding custom hostname %q", newChName)
				_, chErr := p.Client.CreateCustomHostname(ctx, zoneID, change.CustomHostname)
				if chErr != nil {
					failedChange = true
					log.WithFields(logFields).Errorf("failed to add custom hostname %q: %v", newChName, chErr)
				}				
			}
```

Reply:
It's just a suggestion, not a strong opinion, to avoid too many levels.

Wdyt ?

Reply:
definitely my oversight, aggravated by the contrast with:
https://github.com/kubernetes-sigs/external-dns/pull/5183/files#diff-efafdf56d71f1a3f19d003c8bad9fbadb8ee34e873cfa41362ca880e9f7c5fd3R393
just a few lines above

fixed, thanks for pointing it out!